### PR TITLE
test: embedded Kafka broker for integration tests

### LINKED ISSUE #62: Replace testcontainers with "something else"?
I think the testcontainers tests are slow, and we will also be affected by the DockerHub rate limits. There are some suggestions in https://github.com/testcontainers/testcontainers-java/issues/3099, but I personally think we could/should test this without containers. What about an in-memory embedded broker?

### PR DESCRIPTION
**Pull request by sverrehu**
Closes #62 

On my Mac, the time spent inside K3aLagExporterI:

| Branch | ms |
|---|---|
| main | 10.7 |
| sverrehu:embedded-kafka  | 3.6 |
| erikbo:spring-kafka-embedded | 2.6 |

Total runtime for `mvn verify` on GitHub:

| Branch | s |
|---|---|
| main | 42 |
| sverrehu:embedded-kafka  | 12 |
| erikbo:spring-kafka-embedded | 14 |

### CODE REVIEW

File: pom.xml
Comment (erikgb) [inline pom.xml:62]:
```suggestion

        <dependency>
```

For consistency.

File: pom.xml
Comment (erikgb) [inline pom.xml:82]:
```suggestion
```

For consistency.

**Review by erikgb — DISMISSED** [review]
I think this is really cool, and somehow a lot better than the Spring Kafka embedded broker, at least at present. I think we should check Spring Kafka again when support for Kraft is released - since I would prefer to externalize the embedded broker. That will allow us to use the same approach in other projects while maintaining DRY.

Maybe wait with the merge until we get a review from @chromy96?

**Comment by sverrehu** [comment]
Rewrote it to use k3aembedded from Maven Central. PTAL, @erikgb

File: src/test/java/no/statnett/k3alagexporter/itest/K3aLagExporterIT.java
Comment (erikgb) [inline src/test/java/no/statnett/k3alagexporter/itest/K3aLagExporterIT.java:38]:
```suggestion
    private static K3aEmbedded broker;
```

WDYT?

Comment (sverrehu) [reply src/test/java/no/statnett/k3alagexporter/itest/K3aLagExporterIT.java:38]: I'll fix it locally, to do a full rename.

**Review by erikgb — COMMENTED** [review]
Blocked by https://github.com/sverrehu/k3a-embedded/pull/34 and conflicts must be resolved.

**Comment by sverrehu** [comment]
> Blocked by [sverrehu/k3a-embedded#34](https://github.com/sverrehu/k3a-embedded/pull/34) and conflicts must be resolved.

Too bad Maven Central is slow at updating. Conflicts are resolved. Waiting to push the k3aembedded update until it is available, so we don't risk having GitHub cache that the dependency is 404.

File: pom.xml
Comment (erikgb) [inline pom.xml:73]:
```suggestion
            <version>0.2.2+${kafka.version}</version>
```

Comment (sverrehu) [reply pom.xml:73]: Thanks! Wonder if Renovate will handle the semver 2.0.0 plus sign (metadata) correctly.

Comment (erikgb) [reply pom.xml:73]: I think it should. I went through https://semver.org/ to check.

Comment (erikgb) [reply pom.xml:73]: Maybe it does not understand the `${kafka.version}` syntax, as it is a variable. The literal version `0.2.1+${kafka.version}` does not exist.

**Review by erikgb — APPROVED** [review]
Very nice! Good job @sverrehu!

**Pull request merged**
